merkle/src/store.rs

```diff
@@ -287,3 +457,235 @@ impl<E: Element> Clone for DiskStore<E> {
         unimplemented!("We can't clone a store with an already associated file");
     }
 }
+
+
+/// The LevelCacheStore is used to reduce the on-disk footprint even
```

Comment:
can we put this into its own file/each store in its own file, these files are getting out of hand 🗡 

maybe

```
src/store/mod.rs
src/store/disk.rs
src/store/level.rs
src/store/vec.rs
```

Reply:
Definitely, was thinking about something similar, but wanted to save it for the tree re-org (removing the subdir, etc).  Up to you if you want it in this PR.

Reply:
lets do that as a follow up 👌

Reply:
Noting for https://github.com/filecoin-project/merkle_light/issues/45